src/topology/homotopy/H_spaces.lean

```diff
@@ -91,25 +116,32 @@ instance H_space.prod (X : Type u) (Y : Type v) [topological_space X] [topologic
 
 namespace topological_group
 
-@[priority 600, to_additive] instance H_space (G : Type u) [topological_space G] [group G]
-  [topological_group G] : H_space G :=
-{ Hmul := ⟨function.uncurry has_mul.mul, continuous_mul⟩,
+/-- The definition `to_H_space` is not an instance to avoid that the structure `mul_one_class` of
+a topological field creates a diamond. In the case of group, we make `topological_group.H_space` an
+instance.-/
+@[to_additive "The definition `to_H_space` is not an instance to avoid that the structure
+`add_zero_class` of a topological field creates a diamond. In the case of an additive group, we
+make `topological_group.H_space` an instance."]
```

Comment:
Please mention both `mul_one_class` and `add_zero_class` in both docstrings: a topological field is a topological additive group, so it's not clear from the additive docstrings why it can't be an instance.

Reply:
Sure, I have seen and committed both of them, and will also include these fields in both docstrings.